oak_functions/client/java/src/com/google/oak/functions/client/AttestationClient.java

```diff
@@ -45,24 +45,28 @@
 import oak.remote_attestation.ServerIdentity;
 
 /**
- * Client with remote attestation support for sending requests to an Oak Functions loader
- * application.
+ * Client with remote attestation support for sending requests to an Oak Functions application.
  */
 public class AttestationClient {
   private static final Logger logger = Logger.getLogger(AttestationClient.class.getName());
   // TODO(#1867): Add remote attestation support.
   private static final String TEST_TEE_MEASUREMENT = "Test TEE measurement";
-  private final ManagedChannel channel;
-  private final StreamObserver<AttestedInvokeRequest> requestObserver;
-  private final BlockingQueue<AttestedInvokeResponse> messageQueue;
-  private final AeadEncryptor encryptor;
+  private ManagedChannel channel;
+  private StreamObserver<AttestedInvokeRequest> requestObserver;
+  private BlockingQueue<AttestedInvokeResponse> messageQueue;
+  private AeadEncryptor encryptor;
 
   /**
-   * Creates an attested gRPC channel.
+   * Creates an unattested AttestationClient instance.
+   */
+  public AttestationClient() {}
+
+  /**
+   * Creates a gRPC channel and creates an attested channel over it.
    *
-   * `url` must contain a protocol used for connection ("https://" or "http://").
+   * @param url must contain a protocol used for the connection ("https://" or "http://")
    */
-  public AttestationClient(String url)
+  public void attest(String url)
```

Comment:
(FYI) Once we figure out more of how remote attestation will actually work, I think we will want to change this API; we should probably have an AttestationClientFactory or similar, configured with a policy (not just a hardcoded TEE measurement), and with a method to then create an individual attested clients. In that design, the AttestedClient fields would all be final, and the policy verification logic would only live in the factory, not in the established client.

cc @ipetr0v

Reply:
We should also consider a redesign of the lifecycle, e.g. not overriding finalize. See https://google.github.io/styleguide/javaguide.html#s6.4-finalizers